tests/test_config_manager_unit.py

```
@@ -0,0 +1,105 @@
+        ConfigManager.from_file(tmp_bad_schema_file, project_base=tmp_path)
+
+
+def test_validate_and_tool_dispatch(tmp_config_file, tmp_path, mocker):
+    cfg = ConfigManager.from_file(tmp_config_file, project_base=tmp_path)
+    # Patch a validator to check dispatch
+    mock_validator = mocker.patch("utils.validators.mosaic_processor_validator.validate")
+    cfg.TOOL_VALIDATORS["mosaic_processor"] = mock_validator
+    cfg.validate_tool_config("mosaic_processor")
+    mock_validator.assert_called_once_with(cfg)
+    # Unknown tool
+    with pytest.raises(Exception):
```

Ruff (0.11.9)

Do not assert blind exception: `Exception`

(B017)